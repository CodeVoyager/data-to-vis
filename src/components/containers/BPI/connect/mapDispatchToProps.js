import {
    setAvailableCurrencies,
    setCurrency,
    setEndDate,
    setStartDate,
    setData,
    setHighLights,
    setCurrentHighlightDate,
    setCurrentHighlightDescription,
    resetCurrentHighlight,
    addHighLight,
} from '../../../../actions/bpi';
import find from 'ramda/src/find';
import equals from 'ramda/src/equals';
import {
    PREFERED_CURRENCY,
    DATE_FORMAT
} from '../../../../config'
import moment from 'moment';

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        onComponentDidMount: () => {
            dispatch(setHighLights([
                {
                    date: '2006-10-01',
                    description: 'Mt. Gox founding'
                },
                {
                    date: '2015-01-06',
                    description: 'Mt. Gox Bankruptcy'
                },
                {
                    date: '2016-05-01',
                    description: 'Mt. Gox Bankruptcy (Final)'
                },
                {
                    date: '2017-04-01',
                    description: 'Bitfinex announced that it was no longer able to let users withdraw their funds in USD'
                },
            ]));
        },
        onNilStartDate: () => {
            dispatch(setStartDate(moment().subtract(5, 'years')));
        },
        onNilEndDate: () => {
            dispatch(setEndDate(moment()));
        },
        onEndDateChange: (date) => {
            dispatch(setEndDate(date));
        },
        onStartDateChange: (date) => {
            dispatch(setStartDate(date));
        },
        onCurrencyChange: (currency) => {
            dispatch(setCurrency(currency));
        },
        onCurrenciesAvailable: (currencies) => {
            let currency = find(equals(PREFERED_CURRENCY), currencies);

            dispatch(setCurrency(currency || currencies[0]));
            dispatch(setAvailableCurrencies(currencies));
        },
        onDataAvailable: (payload) => {
            dispatch(setData(payload));
        },
        onHighlightDateDescription: (event) => {
            dispatch(setCurrentHighlightDescription(event.target.value));
        },
        onCurrentHighlightDateChange: (payload) => {
            dispatch(setCurrentHighlightDate(payload));
        },
        onCurrentHighlightSubmit: function () {
            dispatch(addHighLight({
                date: this.props.currentHighlight.date.format(DATE_FORMAT),
                description: this.props.currentHighlight.description,
            }));
            dispatch(resetCurrentHighlight());
        }
    }
}

export default mapDispatchToProps;