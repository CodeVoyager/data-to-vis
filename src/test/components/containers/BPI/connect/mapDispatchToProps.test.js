import mapDispatchToProps from '../../../../../components/containers/BPI/connect/mapDispatchToProps';
import moment from 'moment';

describe('BPI connect mapDispatchToProps', () => {

    it('should exist', () => {
        expect(mapDispatchToProps).not.toBeUndefined();
    });


    it('should should pass valid actions dispatch', () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);
        const DATE = moment('20171021', 'YYYYMMDD');
        const CURRENCY = 'USD';
        const CURRENCIES1 = ['USD', 'BGP'];
        const CURRENCIES2 = ['CURRENCY1', 'CURRENCY2'];
        const DATA = [];
        const EVENT = {
            target: {
                value: 'DESCRIPTION'
            }
        };
        const HIGHLIGHT = {
            date: DATE,
            description: EVENT.target.value
        }

        moment.prototype.toJSON = function () {
            return this.format('YYYYMMDD');
        }


        props.onComponentDidMount();
        props.onNilStartDate();
        props.onNilEndDate();
        props.onEndDateChange(DATE);
        props.onStartDateChange(DATE);
        props.onCurrencyChange(CURRENCY);
        props.onCurrenciesAvailable(CURRENCIES1);
        props.onCurrenciesAvailable(CURRENCIES2);
        props.onDataAvailable(DATA);
        props.onHighlightDateDescription(EVENT);
        props.onCurrentHighlightDateChange(HIGHLIGHT);
        props.onCurrentHighlightSubmit.bind({
            props: {
                currentHighlight: HIGHLIGHT
            }
        })(HIGHLIGHT);

        dispatch.mock.calls.forEach((params) => {
            expect(params).toMatchSnapshot();
        });
    });


});
