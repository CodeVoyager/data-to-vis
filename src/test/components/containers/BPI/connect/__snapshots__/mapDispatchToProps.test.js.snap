// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "date": "2006-10-01",
        "description": "Mt. Gox founding",
      },
      Object {
        "date": "2015-01-06",
        "description": "Mt. Gox Bankruptcy",
      },
      Object {
        "date": "2016-05-01",
        "description": "Mt. Gox Bankruptcy (Final)",
      },
      Object {
        "date": "2017-04-01",
        "description": "Bitfinex announced that it was no longer able to let users withdraw their funds in USD",
      },
    ],
    "type": "BPI_SET_HIGHLIGHT",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 2`] = `
Array [
  Object {
    "payload": "20121028",
    "type": "BPI_SET_START_DATE",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 3`] = `
Array [
  Object {
    "payload": "20171028",
    "type": "BPI_SET_END_DATE",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 4`] = `
Array [
  Object {
    "payload": "20171021",
    "type": "BPI_SET_END_DATE",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 5`] = `
Array [
  Object {
    "payload": "20171021",
    "type": "BPI_SET_START_DATE",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 6`] = `
Array [
  Object {
    "payload": "USD",
    "type": "BPI_SET_CURRENCY",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 7`] = `
Array [
  Object {
    "payload": "USD",
    "type": "BPI_SET_CURRENCY",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 8`] = `
Array [
  Object {
    "payload": Array [
      "USD",
      "BGP",
    ],
    "type": "BPI_SET_AVAILABLE_CURRENCIES",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 9`] = `
Array [
  Object {
    "payload": "CURRENCY1",
    "type": "BPI_SET_CURRENCY",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 10`] = `
Array [
  Object {
    "payload": Array [
      "CURRENCY1",
      "CURRENCY2",
    ],
    "type": "BPI_SET_AVAILABLE_CURRENCIES",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 11`] = `
Array [
  Object {
    "payload": Array [],
    "type": "BPI_SET_DATA",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 12`] = `
Array [
  Object {
    "payload": "DESCRIPTION",
    "type": "BPI_SET_CURRENT_HIGHLIGHT_DESCRIPTION",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 13`] = `
Array [
  Object {
    "payload": Object {
      "date": "20171021",
      "description": "DESCRIPTION",
    },
    "type": "BPI_SET_CURRENT_HIGHLIGHT_DATE",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 14`] = `
Array [
  Object {
    "payload": Object {
      "date": "2017-10-21",
      "description": "DESCRIPTION",
    },
    "type": "BPI_ADD_HIGHLIGHT",
  },
]
`;

exports[`BPI connect mapDispatchToProps should should pass valid actions dispatch 15`] = `
Array [
  Object {
    "type": "BPI_SET_CURRENT_HIGHLIGHT_RESET",
  },
]
`;
